jlm/llvm/ir/operators/operators.cpp

```diff
@@ -543,45 +543,46 @@ PoisonValueOperation::copy() const
   return std::make_unique<PoisonValueOperation>(*this);
 }
 
-/* floating point arithmetic operator */
-
-fpbin_op::~fpbin_op()
-{}
+FBinaryOperation::~FBinaryOperation() noexcept = default;
 
 bool
-fpbin_op::operator==(const Operation & other) const noexcept
+FBinaryOperation::operator==(const Operation & other) const noexcept
 {
-  auto op = dynamic_cast<const fpbin_op *>(&other);
+  auto op = dynamic_cast<const FBinaryOperation *>(&other);
   return op && op->fpop() == fpop() && op->size() == size();
 }
 
 std::string
-fpbin_op::debug_string() const
+FBinaryOperation::debug_string() const
 {
-  static std::unordered_map<llvm::fpop, std::string> map({ { fpop::add, "add" },
-                                                           { fpop::sub, "sub" },
-                                                           { fpop::mul, "mul" },
-                                                           { fpop::div, "div" },
-                                                           { fpop::mod, "mod" } });
+  static std::unordered_map<llvm::fpop, std::string> map({ { fpop::add, "Add" },
+                                                           { fpop::sub, "Sub" },
+                                                           { fpop::mul, "Mul" },
```

Comment:
I'm not at a computer so I've not been able to check. 
But I'm guessing these name changes can be the reason for the HLS to fail. The floating-point operations do not get generated. They are instead pre-built macros and I guess this change affects the name so they are not found.

Reply:
Yeah, I figured the same. I changed them back and it seems to do better. This being implicitly bound to the debug string is not great, it should be an explicit mapping. Changing that is for another PR though.